### DISCUSSION TITLE
Error in memcache while trying to use Cortex as a package.

### DISCUSSION THREAD

**Discussion by srijan55** [cortexproject/cortex#4368] [Q&A] [answered]
I am trying to simply use Cortex as a package so I can add some other things before it in my module.

`package main
import "github.com/cortexproject/cortex/pkg/cortex"
func main() {
	cortex.New(cortex.Config{Target: []string{"Frontend"}})
}
`

Getting this error which seems totally unrelated to what I am doing -
`# github.com/cortexproject/cortex/pkg/chunk/cache
C:\Users\manishkg\go\pkg\mod\github.com\cortexproject\cortex@v1.9.0\pkg\chunk\cache\memcached_client.go:144:19: newClient.Client.DialTimeout undefined (type *memcache.Client has no field or method DialTimeout)
exit status 2
Process exiting with code: 1
`

Would appreciate any help/ideas. Thanks!

**Comment by pracucci** [comment, marked as answer]
We have an override in `go.mod`:

```
// Using a 3rd-party branch for custom dialer - see https://github.com/bradfitz/gomemcache/pull/86
replace github.com/bradfitz/gomemcache => github.com/themihai/gomemcache v0.0.0-20180902122335-24332e2d58ab
```

I suggest you to use vendor gomemcache from the same fork.
